src/solvers/refinement/string_refinement.cpp

```diff
@@ -0,0 +1,769 @@
+/** -*- C++ -*- *****************************************************\
+
+Module: String support via axiom instantiation
+        (see the PASS paper at HVC'13)
+
+Author: Alberto Griggio, alberto.griggio@gmail.com
+
+\*******************************************************************/
+
+#include <ansi-c/string_constant.h>
+#include <util/i2string.h>
+#include <util/replace_expr.h>
+#include <solvers/sat/satcheck.h>
+#include <sstream>
+#include <solvers/refinement/string_refinement.h>
+
+// This is mostly for debugging:
+#include <langapi/languages.h>
+#include <ansi-c/ansi_c_language.h>
+
+// Succinct version of pretty()
+std::string string_refinementt::pretty_short(const exprt & expr) {
+  languagest languages(ns, new_ansi_c_language());
+  std::string string_value;
+  languages.from_expr(expr, string_value);
+  return string_value;
+}
```

Comment:
Used `from_expr` in that commit: https://github.com/diffblue/cbmc/pull/341/commits/d39d0142451c725dff0d4c50d9e8975e2607352d